src/hangar/dataloaders/torch_loader.py

```diff
@@ -0,0 +1,118 @@
+        good candidates for being `field_names` and if you haven't provided a good set of
+        `field_names` we'll rename those fields with namedtuples `rename=True`.
+
+    .. note::
+
+        From PyTorch 1.1 onwards, if Dataset returns namedtuple, DataLoader also returns
+        namedtuple
+
+    Parameters
+    ----------
+    hangar_datasets : `hangar.dataset.DatasetDataReader` or a Collection
+        A dataset object, a tuple of dataset object or a list of dataset objects`
+    keys : list or tuple
+        An array of sample names. If given only those samples will fetched from the dataset
+    index_range : slice
+        A python slice object which will be used to find the subset of dataset.
+        Argument `keys` takes priority over `index_range` i.e. if both are given, keys
+        will be used and `index_range` will be ignored
+    field_names : list or tuple of str
+        An array of field names used as the `field_names` for the returned namedtuple. If not
+        given, dataset names will be used as the field_names.
+
+    Examples
+    --------
+    >>> from hangar import Repository
+    >>> from torch.utils.data import DataLoader
+    >>> from hangar.dataloaders import TorchDataset
+    >>> repo = Repository('.')
+    >>> co = repo.checkout()
+    >>> dset = co.datasets['dummy_dset']
+    >>> torch_dset = TorchDataset(dset, index_range=slice(1, 100))
+    >>> loader = DataLoader(torch_dset, batch_size=16)
+    >>> for batch in loader:
+    ...     train_model(batch)
+
+    """
+
+    def __init__(self, hangar_datasets, keys=None, index_range=None, field_names=None):
+        if hasattr(hangar_datasets, 'get') and hasattr(hangar_datasets, 'name'):
+            hangar_datasets = [hangar_datasets]
+
+        try:
+            dataset_len = len(hangar_datasets[0])
+        except IndexError:
+            raise
+        except Exception:
+            raise TypeError("`hangar_datasets` has to be a valid hangar dataset or an"
+                            "iterable that can hold other objects, like a list / tuple etc.")
+
+        dset_names = []
+        for dset in hangar_datasets:
+            if not (hasattr(dset, 'get') and hasattr(dset, 'name')):
+                raise TypeError("`hangar_dataset` contains invalid hangar dataset(s)")
+            if len(dset) != dataset_len:
+                raise RuntimeError("Got datasets with different lengths")
+            dset_names.append(dset.name)
+        self.hangar_datasets = hangar_datasets
+
+        if field_names:
+            if len(field_names) != len(hangar_datasets):
+                raise RuntimeError("length mismatch: `field_names` and `hangar_datasets` must "
+                                   "have same number of elements")
+            self.nt_wrapper = namedtuple('BatchDataset', field_names=field_names)
+        else:
+            self.nt_wrapper = namedtuple('BatchDataset', field_names=dset_names, rename=True)
```

Comment:
Could this be named `'BatchTuple'` instead? It's not a Dataset (technically it won't be a batch either, until after the data loader, so `'SampleTuple'` but since the intended use case is to use a data loader, so it will be correct at that point).

Reply:
That's fine